interfaces/input_parameters/ring.py

```diff
@@ -182,182 +130,348 @@ class Ring:
 
     """
 
-    def __init__(self, ring_length, alpha, Particle,
-                 momentum=None, kin_energy=None, energy=None,
-                 bending_field=None, bending_radius=None,
-                 **kwargs):
-
-        # Checking that at least one synchronous data input is passed
-        syncDataTypes = ('momentum', 'kin_energy', 'energy', 'B_field')
-        syncDataInput = (momentum, kin_energy, energy, bending_field)
-        assrt.single_not_none(*syncDataInput,
-                              msg='Exactly one of '+str(syncDataTypes) +
-                              ' must be declared',
-                              exception=excpt.InputError)
-
-        # Checking that the bending_radius is passed with the bending_field
-        if bending_field is not None and bending_radius is None:
-            raise excpt.InputError("If bending_field is used, bending_radius "
-                                   + "must be defined.")
-
-        # Taking the first synchronous_data input not declared as None
-        # The assertion above ensures that only one is declared
-        for func_type, synchronous_data in zip(syncDataTypes, syncDataInput):
-            if synchronous_data is not None:
-                break
-
-        # Setting ring length, circumerence, radius, bending radius if defined
-        self.ring_length = np.array(ring_length, ndmin=1, dtype=float)
-        self.ring_circumference = np.sum(self.ring_length)
-        self.ring_radius = self.ring_circumference/(2*np.pi)
-
-        if bending_radius is not None:
-            self.bending_radius = float(bending_radius)
-        else:
-            self.bending_radius = bending_radius
+    def __init__(self, Particle, Section_list, **kwargs):
```

Comment:
To me it makes more sense to put the Particle second, but that's personal preference.  What's the purpose of capitalising here?

Reply:
To be checked in the meeting, the reason I put it that way is that I started by declaring 
```python
def __init__(self, Particle, *Section_list, **kwargs)
```
The section list is virtually an *arg list, but since it's not in practice the order of arguments can be changed.

The capitalizing is a thing we used to do from early days of BLonD, to refer that we are expecting a specific type of object directly from the signature. Now that you mention it I am not sure whether there is a PEP about that. Worth investigating!

Reply:
I think it's "wrong" (or as wrong as is possible in a language that allows anything) to use * in front of anything other than 'args' and ** in front of anything other than 'kwargs'.  This is a case where I don't think it's necessary anyway, so the way you're handling it is better.

Be interesting to know what the style guide says here...  The fact they get highlighted by the syntax detection makes me think capitals are specifying _something_, but no idea what.